sync/client_inmem.go

```diff
@@ -0,0 +1,126 @@
+package sync
+
+import (
+	"context"
+	"reflect"
+	"sync"
+)
+
+type inmemClient struct {
+	sync.Mutex
+	*sugarOperations
+
+	states        map[State]int
+	barriers      map[State][]*Barrier
+	subscriptions map[*Topic][]reflect.Value
+	published     map[*Topic][]interface{}
+}
+
+var _ Interface = (*inmemClient)(nil)
+
+// NewInmemClient creates an in-memory sync client for testing.
+func NewInmemClient() Interface {
```

Comment:
I think you could just return the *inmemClient - `accept interfaces, return structs`, although this isn't a hard rule.

Reply:
Yup. I wonder how it would show up in godocs though.

Reply:
![image](https://user-images.githubusercontent.com/1101242/82464907-8982d680-9ab6-11ea-82ad-f5588bc13ae6.png)

... where obviously `*inmemClient` is not documented. I don't think that's nice.